File: lib/java/src/main/java/io/github/telepact/internal/binary/BinaryEncoder.java
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package io.github.telepact.internal.binary;

import java.util.List;

public interface BinaryEncoder {
    List<Object> encode(List<Object> message);

    List<Object> decode(List<Object> message);
}

File: lib/go/internal/binary/BinaryEncoder.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package binary

// BinaryEncoder defines methods for encoding and decoding pseudo-JSON Telepact messages into a binary representation.
type BinaryEncoder interface {
	Encode(message []any) ([]any, error)
	Decode(message []any) ([]any, error)
}
